### PR TITLE
Custom table

### LINKED ISSUE #1654: Custom table

### PR DESCRIPTION
**Pull request by BykhovDenis**
For reviews, `assigned to me`, skills it will be done in future #1732 
close #1654 
Signed-off-by: Denis Bykhov <80476319+BykhovDenis@users.noreply.github.com>

### CODE REVIEW

File: plugins/inventory-resources/src/components/HierarchyView.svelte [hunk 1]
```diff
@@ -20,6 +20,7 @@
   import { buildModel } from '@anticrm/view-resources'
   import { Category } from '@anticrm/inventory'
   import HierarchyElement from './HierarchyElement.svelte'
+import { buildConfigLookup } from '@anticrm/view-resources/src/utils';
```
Comment (haiodo) [inline plugins/inventory-resources/src/components/HierarchyView.svelte]:
```suggestion
  import { buildConfigLookup } from '@anticrm/view-resources/src/utils'
```

File: plugins/inventory-resources/src/components/Products.svelte [hunk 2]
```diff
@@ -20,25 +20,49 @@
...
-  })
+  client
+    .findOne<Viewlet>(view.class.Viewlet, {
+      attachTo: inventory.class.Product,
+      descriptor: view.viewlet.Table
+    })
+    .then((res) => {
+      descr = res
+      if (res !== undefined) {
+        preferenceQuery.query(
```
Comment (haiodo) [inline plugins/inventory-resources/src/components/Products.svelte:53]: May be we could query with lookup to automatically view.class.ViewletPreference with Viewlet inside?

File: plugins/tracker-resources/src/components/issues/IssuesList.svelte [hunk 3]
```diff
@@ -30,6 +30,7 @@
   } from '@anticrm/ui'
   import { AttributeModel, BuildModelKey } from '@anticrm/view'
   import { buildModel, getObjectPresenter, LoadingProps, Menu } from '@anticrm/view-resources'
+import { buildConfigLookup } from '@anticrm/view-resources/src/utils';
```
Comment (haiodo) [inline plugins/tracker-resources/src/components/issues/IssuesList.svelte]:
```suggestion
  import { buildConfigLookup } from '@anticrm/view-resources/src/utils'
```

File: plugins/view-resources/src/components/ViewletSetting.svelte [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  function getConfig (viewlet: Viewlet, preference: ViewletPreference | undefined): AttributeConfig[] {
+    const result = getBaseConfig(viewlet)
+
+    const allAttributes = hierarchy.getAllAttributes(viewlet.attachTo)
+    for (const [, attribute] of allAttributes) {
+      const value = hierarchy.isDerived(attribute.type._class, core.class.RefTo)
+        ? '$lookup.' + attribute.name
+        : attribute.name
+      if (result.findIndex((p) => p.value === value) !== -1) continue
+      if (hierarchy.isDerived(attribute.type._class, core.class.Collection)) continue
```
Comment (haiodo) [inline plugins/view-resources/src/components/ViewletSetting.svelte:107]: How ArrOf(TypeRef())) is handled?

**Review by haiodo — COMMENTED** [review]
Look nice.

File: plugins/inventory-resources/src/components/Products.svelte (hunk 2 above)
Comment (BykhovDenis) [reply plugins/inventory-resources/src/components/Products.svelte:53]: Preference should be in LQ, viewlet - no. I can put it all in LQ

Comment (BykhovDenis) [reply plugins/inventory-resources/src/components/Products.svelte:53]: We can't lookup model (Viewlet) with ServerStorage(Preference)

File: plugins/view-resources/src/components/ViewletSetting.svelte (hunk 4 above)
Comment (BykhovDenis) [reply plugins/view-resources/src/components/ViewletSetting.svelte:107]: Fixed

**Review by haiodo — APPROVED** [review]

**Pull request merged**